src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.ts

```diff
@@ -213,10 +220,17 @@ export class ServiceFormComponent extends CdForm implements OnInit {
   }
 
   ngOnInit(): void {
-    if (this.router.url.includes('services')) {
+    this.action = this.actionLabels.CREATE;
+    if (this.router.url.includes('services/(modal:create')) {
+      this.pageURL = 'services';
+    } else if (this.router.url.includes('services/(modal:edit')) {
+      this.editing = true;
       this.pageURL = 'services';
+      this.route.params.subscribe((params: { type: string; name: string }) => {
+        this.serviceName = params.name;
+        this.serviceType = params.type;
+      });
```

Comment:
I was recently thinking if there would be a way to pass router params as component `@Inputs`... And found [this library](https://www.npmjs.com/package/ngx-route-params-input) and a [blog post explaining how it works](https://medium.com/@andrewcherepovskiy/passing-route-params-into-angular-components-input-properties-fc85c34c9aca).